pcdsdevices/variety.py

```diff
@@ -70,83 +70,166 @@ def validate(value):
+        Optional('value', default=1): schema.Or(float, int, str),
+        Optional('enum_strings'): [str],
+        Optional('enum_dict'): dict,
         **common_schema
     }),
 
     'array': schema.Schema({
         'variety': schema.Or(*varieties_by_category['array']),
-        schema.Optional('shape'): _length_validate(1, 10, int),
-        schema.Optional('dimension'): int,
-        schema.Optional('embed'): bool,
-        schema.Optional('colormap'): str,
+        Optional('shape'): _length_validate(1, 10, int),
+        Optional('shape_signal'): str,
+        Optional('dimension'): int,
+        Optional('embed'): bool,
+        Optional('colormap'): str,
         **common_schema
     }),
 
     'bitmask': schema.Schema({
         'variety': schema.Or(*varieties_by_category['bitmask']),
-        schema.Optional('orientation',
-                        default='horizontal'): schema.Or('horizontal',
-                                                         'vertical'),
-        schema.Optional('bits', default=8): int,
-        schema.Optional(
-            'first_bit',
-            default='most-significant'): schema.Or('most-significant',
-                                                   'least-significant'),
-
-        # Style:
-        schema.Optional('shape', default='rectangle'): schema.Or('circle',
-                                                                 'rectangle'),
-        schema.Optional('on_color', default='green'): str,
-        schema.Optional('off_color', default='gray'): str,
+        Optional('orientation', default='horizontal'): schema.Or(
+            'horizontal', 'vertical'),
+        Optional('bits', default=8): int,
+        Optional('first_bit', default='most-significant'): schema.Or(
+            'most-significant', 'least-significant'),
+        Optional('meaning', default=None): [str],
+
+        Optional('style', default=_default_bitmask_style): schema.Schema({
+            Optional('shape', default='rectangle'): schema.Or(
+                'circle', 'rectangle'),
+
+            Optional('on_color', default='green'): str,
+            Optional('off_color', default='gray'): str,
+        }),
         **common_schema
     }),
 
     'scalar': schema.Schema({
         'variety': schema.Or(*varieties_by_category['scalar']),
-        schema.Optional('range'): _length_validate(2, 2, (float, int)),
-        schema.Optional('range_source'): schema.Or('use_limits', 'custom'),
-        schema.Optional('display_format', default='default'): schema.Or(
+        Optional('display_format', default='default'): schema.Or(
             'default', 'string', 'decimal', 'exponential', 'hex', 'binary'),
+
+        Optional('range'): schema.Schema({
+            Optional('value'): _length_validate(2, 2, (float, int)),
+            Optional('source', default='use_limits'): schema.Or(
+                'use_limits', 'value'),
+        }),
+
+        Optional('delta'): schema.Schema({
```

Comment:
I'm having trouble interpreting what a scalar's delta would be. Can you give an example?

Reply:
The full name would be "delta [tweak] value". So, basically what is a reasonable "delta" to change the scalar by? And what is a valid range of that delta?

A sample value would be for a 0-100% value, a standard delta of 0.1%, with an adjustable range of `-1% <= delta <= 1%` if it's for some reason advisable to not tweak by larger than 1% (🤷 maybe unnecessary?)

Reply:
Gotcha